### DISCUSSION TITLE
Limit grid import

### DISCUSSION THREAD

**Discussion by djfanatix** [springfall2008/batpred#1196] [General]
In Belgium (and Norway too) we have to pay extra when you have a high grid import. (Capacity tariff)
This is measured by the highest peak per month during a 15 minutes period.

Is there a possibility to have a setting to limit grid import? (For example when charging battery from grid max 2500w, but no limit when charging from solar?)

thanks

**Comment by gcoan** [comment]
I'd suggest you raise this as a [feature request](https://github.com/springfall2008/batpred/issues/new?assignees=&labels=&projects=&template=feature_request.md&title=) rather than a discussion item, its more likely to be seen there.

Sounds like it should be achievable, in fact I'd like to see it extended so you can use a Home Assistant entity (not just a fixed value) to control both grid and battery charge rate, that way it could also be used to manage inverter clipping as well

**Reply by djfanatix** [reply]
Ok will do.

I can agree on to use a HA entity, but I don't think it can be combined with battery charge rate.

For example I have a SolarEdge installation, and if the parameter for battery charging is limited, this will be limited when charging from the grid, but also from solar.
So I think this has to be 2 separate entities.
